### PR TITLE
Allow config to have multiple remotes.

### PR DESCRIPTION
**Pull request by markgardner**
This will enable development teams to test emails from a system like mailcatcher while also having the email delivered to the indented mailbox.

### CODE REVIEW

**Comment by JonathonReinhart** [comment]
I haven't looked closely yet, but this has some overlap with (and if accepted, might supersede) #31.

**Comment by markgardner** [comment]
Agreed, the scheme will allow TLS flow to be forced

File: remotes.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+// - auth: can be "login" to trigger "LOGIN" auth instead of "PLAIN" auth
+//
+func ParseRemote(remoteURL string) (*Remote, error) {
+	u, err := url.Parse(remoteURL)
+	if err != nil {
+		return nil, err
+	}
+
+	hostname, port := u.Hostname(), u.Port()
+
```
Comment (decke) [inline remotes.go:35]: There could be more cases ...

if u.Scheme == "" {
default smtp://
}

if port == "" {
smtp:// => default 25
smtps:// => default 465
starttls:// => default 587
}

File: remotes.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	Port       string
+	Addr       string
+	Sender     string
+}
+
+// ParseRemote creates a remote from a given url in the following format:
+//
+// smtp://[user[:password]@][netloc][:port][/remote_sender][?param1=value1&...]
+// smtps://[user[:password]@][netloc][:port][/remote_sender][?param1=value1&...]
+//
```
Comment (decke) [inline remotes.go:23]: + starttls://

File: smtprelay.ini [hunk 3]
```diff
@@ -87,27 +87,25 @@
 ; If not set, mails are discarded.
 
 ; GMail
-;remote_host = smtp.gmail.com:587
+;remotes = smtp://user:pass@smtp.gmail.com:587
 
 ; Mailgun.org
-;remote_host = smtp.mailgun.org:587
+;remotes = smtp://user:pass@smtp.mailgun.org:587
```
Comment (decke) [inline smtprelay.ini:93]: with the above we could then change these lines to starttls://

File: smtp.go [hunk 4]
```diff
@@ -331,30 +335,27 @@ func SendMail(addr string, a smtp.Auth, from string, to []string, msg []byte) er
...
 		defer conn.Close()
-		c, err = NewClient(conn, host)
+		c, err = NewClient(conn, r.Hostname)
 		if err != nil {
 			return err
 		}
 		if err = c.hello(); err != nil {
 			return err
 		}
 	} else {
```
Comment (decke) [inline smtp.go:357]: So here it could be smtp:// or starttls://

File: smtp.go [hunk 5]
```diff
@@ -375,11 +376,11 @@ func SendMail(addr string, a smtp.Auth, from string, to []string, msg []byte) er
 			}
 		}
```
Comment (decke) [inline smtp.go:377]: We should check here if scheme is starttls:// but server did not announce STARTTLS extension => raise an error

**Comment by decke** [comment]
The change is really great, thanks a lot for your work!

File: smtprelay.ini [hunk 6]
```diff
@@ -87,27 +87,25 @@
 ; If not set, mails are discarded.
 
 ; GMail
-;remote_host = smtp.gmail.com:587
+;remotes = smtp://user:pass@smtp.gmail.com:587
```
[the remark this answers is not in the document]
Comment (markgardner) [reply smtprelay.ini]: thanks, that's a good call. I've updated the examples to use starttls

File: remotes.go (hunk 2 above)
Comment (markgardner) [reply remotes.go:24]: added

File: smtp.go (hunk 5 above)
Comment (markgardner) [reply smtp.go:379]: Good call, I've added the error

**Comment by markgardner** [comment]
@decke I've made the changes. Let me know if there is anything else you see.

Thanks for the project, this helps us a ton with testing.

File: smtp.go (hunk 5 above)
Comment (decke) [reply smtp.go:379]: This diff looks a bit strange on github (don't know why) but I will have a look at that after the merge. Thanks!

**Review by decke — APPROVED** [review]

**Comment by decke** [comment]
Thanks a lot for your contribution! :o)

**Pull request merged**